### DISCUSSION TITLE
compile fails with current pull

### DISCUSSION THREAD

**Discussion by ryanpg123** [ElectroTechnique/electrotechnique.github.io#84] [Software Issues]
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/effect_ensemble.cpp: In member function 'int16_t AudioEffectEnsemble::interpBuffer(float)':
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/effect_ensemble.cpp:267:16: error: ordered comparison of pointer with integer zero ('char* (*)(const char*, int)' and 'int')
  267 |   else if(index<0) index2+=ENSEMBLE_BUFFER_SIZE;
      |           ~~~~~^~
In file included from /home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino:57:
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/AudioPatching.h:190:5: error: 'AudioOutputUSB' does not name a type; did you mean 'AudioOutputI2S'?
  190 |     AudioOutputUSB           usbAudio;
      |     ^~~~~~~~~~~~~~
      |     AudioOutputI2S
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/AudioPatching.h:216:28: error: 'usbAudio' was not declared in this scope
  216 |         {effectMixerRM, 0, usbAudio, 1},
      |                            ^~~~~~~~
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/AudioPatching.h:219:28: error: 'usbAudio' was not declared in this scope
  219 |         {effectMixerLM, 0, usbAudio, 0}
      |                            ^~~~~~~~
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/AudioPatching.h:220:5: error: could not convert '{((Global*)this)->Global::effectMixerRM, 0, <expression error>, 1}' from '<brace-enclosed initializer list>' to 'AudioConnection'
  220 |     };
      |     ^
      |     |
      |     <brace-enclosed initializer list>
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/AudioPatching.h:220:5: error: could not convert '{((Global*)this)->Global::effectMixerLM, 0, <expression error>, 0}' from '<brace-enclosed initializer list>' to 'AudioConnection'
  220 |     };
      |     ^
      |     |
      |     <brace-enclosed initializer list>
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino: In function 'void setup()':
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino:185:3: error: 'usbMIDI' was not declared in this scope; did you mean 'MIDI'?
  185 |   usbMIDI.setHandleControlChange(myControlChange);
      |   ^~~~~~~
      |   MIDI
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino: In function 'void midiCCOut(byte, byte)':
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino:1790:5: error: 'usbMIDI' was not declared in this scope; did you mean 'MIDI'?
 1790 |     usbMIDI.sendControlChange(cc, value, midiOutCh);
      |     ^~~~~~~
      |     MIDI
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino: In function 'void loop()':
/home/ryanpg/src/TSynth-Teensy4.1/TSynth/TSynth.ino:1815:3: error: 'usbMIDI' was not declared in this scope; did you mean 'MIDI'?
 1815 |   usbMIDI.read(midiChannel);
      |   ^~~~~~~
      |   MIDI

exit status 1

Compilation error: ordered comparison of pointer with integer zero ('char* (*)(const char*, int)' and 'int')

**Comment by ryanpg123** [comment]
Fixed this by Tools > USB Type = "Serial + MIDI + Audio"

**Comment by Electrotechnique** [comment]
Fixed in V2.35
